soc/xtensa/intel_adsp/common/mem_window.c

```diff
@@ -36,56 +36,27 @@ __imr int mem_win_init(const struct device *dev)
 	return 0;
 }
 
-#if DT_NODE_HAS_STATUS(DT_NODELABEL(mem_window0), okay)
-static const struct mem_win_config mem_win_config_0 = {
-	.base_addr = DT_REG_ADDR(DT_NODELABEL(mem_window0)),
-	.size = WIN_SIZE(0),
-	.offset = WIN_OFFSET(0),
-	.read_only = DT_PROP(DT_NODELABEL(mem_window0), read_only),
-	.mem_base = DT_REG_ADDR(DT_PHANDLE(DT_NODELABEL(mem_window0), memory)) + WIN_OFFSET(0),
-	.initialize = DT_PROP(DT_NODELABEL(mem_window0), initialize),
-};
-DEVICE_DT_DEFINE(DT_NODELABEL(mem_window0), mem_win_init, NULL, NULL, &mem_win_config_0, ARCH,
-		 CONFIG_KERNEL_INIT_PRIORITY_DEFAULT, NULL);
+#define MEM_WINDOW_DEFINE(n)                                                                       \
+	static const struct mem_win_config mem_win_config_##n = {                                  \
+		.base_addr = DT_REG_ADDR(MEM_WINDOW_NODE(n)),                                      \
+		.size = WIN_SIZE(n),                                                               \
+		.offset = WIN_OFFSET(n),                                                           \
+		.read_only = DT_PROP(MEM_WINDOW_NODE(n), read_only),                               \
+		.mem_base = DT_REG_ADDR(DT_PHANDLE(MEM_WINDOW_NODE(n), memory)) + WIN_OFFSET(n),   \
+		.initialize = DT_PROP(MEM_WINDOW_NODE(n), initialize),                             \
+	};                                                                                         \
+	DEVICE_DT_DEFINE(MEM_WINDOW_NODE(n), mem_win_init, NULL, NULL, &mem_win_config_##n, ARCH,  \
+			 CONFIG_KERNEL_INIT_PRIORITY_DEFAULT, NULL);
+
+#if DT_NODE_HAS_STATUS(MEM_WINDOW_NODE(0), okay)
+MEM_WINDOW_DEFINE(0)
 #endif
-
-#if DT_NODE_HAS_STATUS(DT_NODELABEL(mem_window1), okay)
-static const struct mem_win_config mem_win_config_1 = {
-	.base_addr = DT_REG_ADDR(DT_NODELABEL(mem_window1)),
-	.size = WIN_SIZE(1),
-	.offset = WIN_OFFSET(1),
-	.read_only = DT_PROP(DT_NODELABEL(mem_window1), read_only),
-	.mem_base = DT_REG_ADDR(DT_PHANDLE(DT_NODELABEL(mem_window1), memory)) + WIN_OFFSET(1),
-	.initialize = DT_PROP(DT_NODELABEL(mem_window1), initialize),
-};
-DEVICE_DT_DEFINE(DT_NODELABEL(mem_window1), mem_win_init, NULL, NULL, &mem_win_config_1, ARCH,
-		 CONFIG_KERNEL_INIT_PRIORITY_DEFAULT, NULL);
+#if DT_NODE_HAS_STATUS(MEM_WINDOW_NODE(1), okay)
+MEM_WINDOW_DEFINE(1)
```

Comment:
Should be squashed into the previous commit. Otherwise from a DT/driver side this all looks very reasonable to me.